dask/array/core.py

```diff
@@ -817,7 +817,7 @@ def numblocks(self):
 
     @property
     def npartitions(self):
-        return np.prod(self.numblocks)
+        return reduce(mul, self.numblocks, 1)
```

Comment:
Might be worth defining a `product` utility function rather than repeating this all over the place.